tests/integration-tests/tests/networking/test_networking.py

```diff
@@ -0,0 +1,188 @@
+    check_nat_in_subnet_route(ec2_client, private_subnet_id)
+
+
+def check_internet_gateway_in_subnet_route(ec2_client, subnet_id, expected_internet_gateway_id=""):
+    """
+    Check that the given internet_gateway is associated with the route of the subnet.
+
+    :param ec2_client: the boto3 client to which make requests
+    :param subnet_id: the subnet associated with the route we want to verify
+    :param expected_internet_gateway_id: the gateway we expect to find in the route
+    """
+    response = ec2_client.describe_route_tables(Filters=[{"Name": "association.subnet-id", "Values": [subnet_id]}])
+    routes = response["RouteTables"][0]["Routes"]
+    # Routes[1] because 0 is always local
+    assert_that(routes[1]["DestinationCidrBlock"]).is_equal_to("0.0.0.0/0")
+    if expected_internet_gateway_id:
+        assert_that(routes[1]["GatewayId"]).is_equal_to(expected_internet_gateway_id)
+
+
+def check_subnet_cidr(ec2_client, subnet_id, expected_subnet_cidr):
+    """Check that the given subnet has the same subnet cidr."""
+    response = ec2_client.describe_subnets(Filters=[{"Name": "subnet-id", "Values": [subnet_id]}])
+    subnet_cidr = response["Subnets"][0]["CidrBlock"]
+    assert_that(subnet_cidr).is_equal_to(expected_subnet_cidr)
+
+
+def check_internet_gateway_id(ec2_client, vpc_id, expected_internet_gateway_id=""):
+    """Check that the vpc contains the given internet gateway."""
+    if expected_internet_gateway_id:
+        response = ec2_client.describe_internet_gateways(Filters=[{"Name": "attachment.vpc-id", "Values": [vpc_id]}])
+        internet_gateway_id = response["InternetGateways"][0]["InternetGatewayId"]
+        assert_that(internet_gateway_id).is_equal_to(expected_internet_gateway_id)
+
+
+def check_subnet_property(region, subnet_id, expected_autoassign_ip_value, expected_availability_zone=""):
+    """Check that the subnet has the given property."""
+    subnet = boto3.resource("ec2", region_name=region).Subnet(subnet_id)
+    assert_that(subnet.map_public_ip_on_launch).is_equal_to(expected_autoassign_ip_value)
+    if expected_availability_zone:
+        assert_that(subnet.availability_zone).is_equal_to(expected_availability_zone)
+
+
+def check_nat_in_subnet(ec2_client, subnet_id):
+    """Check that there is a nat in the given subnet."""
+    response = ec2_client.describe_nat_gateways(Filters=[{"Name": "subnet-id", "Values": [subnet_id]}])
+    assert_that(len(response["NatGateways"])).is_greater_than(0)
+
+
+def check_nat_in_subnet_route(ec2_client, subnet_id):
+    """Check that the route of the given subnet contains a Nat Gateway."""
+    response = ec2_client.describe_route_tables(Filters=[{"Name": "association.subnet-id", "Values": [subnet_id]}])
+    routes = response["RouteTables"][0]["Routes"]
+    assert_that(
+        "NatGatewayId" in next(route for route in routes if route["DestinationCidrBlock"] == "0.0.0.0/0")
+    ).is_true()
+
+
+def ec2_conn(region):
```

Comment:
probably we can get rid of this since 

ec2_conn(region) is the same as boto3.client("ec2", region_name=region)

and the second one is more explicit